### PR TITLE
Add dkim support

### PR DESCRIPTION
**Pull request by bityob**

### CODE REVIEW

**Review by kootenpv — COMMENTED** [review]
Thanks a lot!

Just wondering, why not make it a function rather than a class?

`def add_dkim_sig_to_message(domain, private_key, include_headers, selector)`?

**Comment by bityob** [comment]
@kootenpv Sure, will convert it to a function instead of class

**Comment by bityob** [comment]
Regarding the namedtuple class, it makes sense to have a data class for all dkim properties together, and not add all the variables detailed on the SMTP class

**Comment by kootenpv** [comment]
Looks good to me, is it ready to be merged?

**Comment by bityob** [comment]
@kootenpv Yes, it's ready. Just need to bump the package number etc. 

Is it part of this PR or you are doing it in another PR?

**Comment by bityob** [comment]
- Added dkimpy as dependency 
- Updated version to `0.15.0`
- Added missing supported python versions to `tox`

File: setup.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
     author_email='kootenpv@gmail.com',
     license='MIT',
     extras_require={"all": ["keyring"]},
-    install_requires=["premailer"],
+    install_requires=["premailer", "dkimpy"],
```
Comment (kootenpv) [inline setup.py:21]: I think we better make it an extras require as I don't think this should be there by default.

```
-    extras_require={"all": ["keyring"]},
+    extras_require={"all": ["keyring", "dkim"], "dkim": ["dkim"]},
```

I think we can just try to 

```python
try:
    import dkim
except ImportError:
   pass
```

In the readme we can show how to install:

    pip install yagmail[dkim]

**Review by kootenpv — COMMENTED** [review]
Thanks a lot and sorry it's been slower, being at in-laws...

maybe could also add entries in .travis.yml

note: there is a bump script but manual is fine too

File: setup.py (hunk 1 above)
Comment (bityob) [reply setup.py:21]: ok, good idea, will add

**Comment by bityob** [comment]
@kootenpv updated, please review again :pray:

**Comment by kootenpv** [comment]
@bityob Thanks a lot for your great contribution. I'm really sorry it took so long, but I verified it to be working. Excellent job!

**Comment by kootenpv** [comment]
You can grab your new version by

    pip install -U --no-cache yagmail>=0.15.277

(the .277 part is the commit count, my script automatically changes that even though you committed 0.15.0)

**Pull request merged**